src/main/java/com/incognia/IncogniaAPI.java

```diff
@@ -28,14 +25,25 @@ public IncogniaAPI(String clientId, String clientSecret, Region region) {
     Asserts.assertNotEmpty(clientId, "client id");
     Asserts.assertNotEmpty(clientSecret, "client secret");
     Asserts.assertNotEmpty(apiUrl, "api url");
-    this.clientId = clientId;
-    this.clientSecret = clientSecret;
-    this.apiUrl = apiUrl;
     // TODO (rato): set client timeout
-    this.httpClient = new OkHttpClient.Builder().build();
+    tokenAwareNetworkingClient =
+        new TokenAwareNetworkingClient(
+            new OkHttpClient.Builder().build(), apiUrl, clientId, clientSecret);
   }
 
-  // POST onboarding
+  public SignupResponse registerSignup(String installationId, Address address)
+      throws IncogniaException {
+    Asserts.assertNotEmpty(installationId, "installation id");
+    Asserts.assertNotNull(address, "address");
```

Comment:
Should we make Address mandatory? Although it is a recommended field, it is currently optional in our docs (and one could easily create an empty Address)

Reply:
It would be quite weird to have a signup without an address. If the user really wants it, they could instantiate an empty address.